Panic in outputs.heartbeat plugin since v1.38.0

### Relevant telegraf.conf

```toml
# Global tags can be specified here in key="value" format.
[global_tags]

# Configuration for telegraf agent
[agent]
  interval = "10s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "0s"
  flush_interval = "10s"
  flush_jitter = "0s"
  precision = "0s"

# A plugin that can transmit heartbeats over HTTP
[[outputs.heartbeat]]
  ## URL of heartbeat endpoint
  url = "http://monitoring.example.com/heartbeat"

  ## Unique identifier to submit for the Telegraf instance (required)
  instance_id = "agent-123"

  ## Token for bearer authentication
  # token = ""

  ## Interval for sending heartbeat messages
  # interval = "1m"

# Read metrics about cpu usage
[[inputs.cpu]]
  ## Whether to report per-cpu stats or not
  percpu = true
  ## Whether to report total system cpu stats or not
  totalcpu = true
  ## If true, collect raw CPU time metrics
  collect_cpu_time = false
  ## If true, compute and report the sum of all non-idle CPU states
  ## NOTE: The resulting 'time_active' field INCLUDES 'iowait'!
  report_active = false
  ## If true and the info is available then add core_id and physical_id tags
  core_tags = false


# Read metrics about temperature
[[inputs.temp]]
  ## Desired output format (Linux only)
  ## Available values are
  ##   v1 -- use pre-v1.22.4 sensor naming, e.g. coretemp_core0_input
  ##   v2 -- use v1.22.4+ sensor naming, e.g. coretemp_core_0_input
  # metric_format = "v2"

  ## Add device tag to distinguish devices with the same name (Linux only)
  # add_device_tag = false
```

### Logs from Telegraf

```text
2026-03-18T09:35:58Z W! Strict environment variable handling is the new default starting with v1.38.0! If your configuration does not work with strict handling please explicitly add the --non-strict-env-handling flag to switch to the previous behavior!
2026-03-18T09:35:58Z I! Loading config: telegraf_test.conf
2026-03-18T09:35:58Z I! Starting Telegraf 1.38.1 brought to you by InfluxData the makers of InfluxDB
2026-03-18T09:35:58Z I! Available plugins: 244 inputs, 9 aggregators, 35 processors, 26 parsers, 68 outputs, 8 secret-stores
2026-03-18T09:35:58Z I! Loaded inputs: cpu temp
2026-03-18T09:35:58Z I! Loaded aggregators:
2026-03-18T09:35:58Z I! Loaded processors:
2026-03-18T09:35:58Z I! Loaded secretstores:
2026-03-18T09:35:58Z I! Loaded outputs: heartbeat
2026-03-18T09:35:58Z I! Tags enabled: host=louis-laptop
2026-03-18T09:35:58Z I! [agent] Config: Interval:1s, Quiet:false, Hostname:"louis-laptop", Flush Interval:10s
2026-03-18T09:35:58Z W! [agent] The default value of 'skip_processors_after_aggregators' will change to 'true' with Telegraf v1.40.0! If you need the current default behavior, please explicitly set the option to 'false'!
2026-03-18T09:35:58Z D! [agent] Initializing plugins
2026-03-18T09:35:58Z D! [agent] Connecting outputs
2026-03-18T09:35:58Z D! [agent] Attempting connection to [outputs.heartbeat]
2026-03-18T09:35:58Z D! [agent] Successfully connected to outputs.heartbeat
2026-03-18T09:35:58Z D! [agent] Starting service inputs
2026-03-18T09:36:08Z D! [outputs.heartbeat] Wrote batch of 371 metrics in 789ns
2026-03-18T09:36:08Z D! [outputs.heartbeat] Buffer fullness: 0 / 10000 metrics
panic: interface conversion: interface {} is nil, not int64

goroutine 114 [running]:
github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*statistics).snapshot(0xc000e7f5c8)
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/statistics.go:86 +0xa78
github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*Heartbeat).send(0xc000e7f208)
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/heartbeat.go:246 +0x52
github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*Heartbeat).Connect.func1({0xb9805d0, 0xc001a12050})
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/heartbeat.go:204 +0x118
created by github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*Heartbeat).Connect in goroutine 1
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/heartbeat.go:192 +0x2d0
```

### System info

Telegraf 1.38.1, Fedora 43 (amd64) and Ubuntu 22 (arm64)

### Docker

_No response_

### Steps to reproduce

1. Create a simple config with the heartbeat output and simple inputs like CPU or Temps
```
~/Downloads/telegraf-1.38.1 config --output-filter "heartbeat" --input-filter "cpu:temp" --aggregator-filter "none" --processor-filter "none" --secretstore-filter "none" --strict-env-handling > telegraf_test.conf
```

2. (Optional) modify the heartbeat interval to wait less
3. Launch Telegraf
```
~/Downloads/telegraf-1.38.1 --config telegraf_test.conf --debug
```


### Expected behavior

Fail because monitoring.example.com host not found

2026-03-18T09:35:18Z E! [outputs.heartbeat] sending message failed: Post "http://monitoring.example.com/heartbeat": dial tcp: lookup monitoring.example.com on 127.0.0.53:53: no such host

### Actual behavior

2026-03-18T09:35:58Z W! Strict environment variable handling is the new default starting with v1.38.0! If your configuration does not work with strict handling please explicitly add the --non-strict-env-handling flag to switch to the previous behavior!
2026-03-18T09:35:58Z I! Loading config: telegraf_test.conf
2026-03-18T09:35:58Z I! Starting Telegraf 1.38.1 brought to you by InfluxData the makers of InfluxDB
2026-03-18T09:35:58Z I! Available plugins: 244 inputs, 9 aggregators, 35 processors, 26 parsers, 68 outputs, 8 secret-stores
2026-03-18T09:35:58Z I! Loaded inputs: cpu temp
2026-03-18T09:35:58Z I! Loaded aggregators:
2026-03-18T09:35:58Z I! Loaded processors:
2026-03-18T09:35:58Z I! Loaded secretstores:
2026-03-18T09:35:58Z I! Loaded outputs: heartbeat
2026-03-18T09:35:58Z I! Tags enabled: host=louis-laptop
2026-03-18T09:35:58Z I! [agent] Config: Interval:1s, Quiet:false, Hostname:"louis-laptop", Flush Interval:10s
2026-03-18T09:35:58Z W! [agent] The default value of 'skip_processors_after_aggregators' will change to 'true' with Telegraf v1.40.0! If you need the current default behavior, please explicitly set the option to 'false'!
2026-03-18T09:35:58Z D! [agent] Initializing plugins
2026-03-18T09:35:58Z D! [agent] Connecting outputs
2026-03-18T09:35:58Z D! [agent] Attempting connection to [outputs.heartbeat]
2026-03-18T09:35:58Z D! [agent] Successfully connected to outputs.heartbeat
2026-03-18T09:35:58Z D! [agent] Starting service inputs
2026-03-18T09:36:08Z D! [outputs.heartbeat] Wrote batch of 371 metrics in 789ns
2026-03-18T09:36:08Z D! [outputs.heartbeat] Buffer fullness: 0 / 10000 metrics
panic: interface conversion: interface {} is nil, not int64

goroutine 114 [running]:
github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*statistics).snapshot(0xc000e7f5c8)
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/statistics.go:86 +0xa78
github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*Heartbeat).send(0xc000e7f208)
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/heartbeat.go:246 +0x52
github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*Heartbeat).Connect.func1({0xb9805d0, 0xc001a12050})
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/heartbeat.go:204 +0x118
created by github.com/influxdata/telegraf/plugins/outputs/heartbeat.(*Heartbeat).Connect in goroutine 1
        /go/src/github.com/influxdata/telegraf/plugins/outputs/heartbeat/heartbeat.go:192 +0x2d0

### Additional info

The heartbeat default config was working fine in Telegraf 1.37.3 (but the features were way more basic).

Maybe my config is too simple or I missed a critical parameter.

Let me know if something isn't clear or if you need more informations.

--- a/models/running_input.go
+++ b/models/running_input.go
@@ -76,7 +76,7 @@ func NewRunningInput(input telegraf.Input, config *InputConfig) *RunningInput {
 			tags,
 		),
 		StartupErrors: selfstat.Register(
-			"write",
+			"gather",
 			"startup_errors",
 			tags,
 		),

